packages/xpath/src/functions/xforms/node-set.ts

```diff
@@ -384,8 +386,39 @@ export const randomize = new NodeSetFunction(
 
 		const nodeResults = Array.from(results.values());
 		const nodes = nodeResults.map(({ value }) => value);
-		const seed = seedExpression?.evaluate(context).toNumber();
-
-		return seededRandomize(nodes, seed);
+		if (seedExpression === undefined) return seededRandomize(nodes);
+		const seed = seedExpression.evaluate(context);
+		const asNumber = seed.toNumber(); // TODO: There are some peculiarities to address: https://github.com/getodk/web-forms/issues/240
```

Comment:
I'm not sure this comment belongs here. It isn't specific to this cast, it's specific to casting to XPath `number` throughout. Fine to leave since we have an issue tracking it, but we'll probably just find it went stale some time after we address the issue.